native/src/main/java/io/ballerina/stdlib/data/xmldata/FromString.java

```diff
@@ -73,6 +92,8 @@ private static Object fromStringWithType(BString string, Type expType) {
                     return stringToNull(value);
                 case TypeTags.UNION_TAG:
                     return stringToUnion(string, (UnionType) expType);
+                case TypeTags.JSON_TAG:
+                    return stringToUnion(string, JSON_TYPE_WITH_BASIC_TYPES);
```

Comment:
Shall we add a test with `json` expected type to cover this

Reply:
Added [9c9b207](https://github.com/ballerina-platform/module-ballerina-data-xmldata/pull/5/commits/9c9b207c80dddcdc4f05fe5aa07c5303664cba6a)